tianshou/policy/imitation/cql.py

```diff
@@ -251,6 +255,23 @@ def learn(self, batch: RolloutBatchProtocol, *args: Any, **kwargs: Any) -> dict[
         ]:
             value.reshape(batch_size, self.num_repeat_actions, 1)
 
+        if self.calibrated:
+            returns = (
+                batch.calibration_returns.unsqueeze(1)
+                .repeat(
+                    (1, self.num_repeat_actions),
+                )
+                .view(-1, 1)
+            )
+            random_value1 = torch.max(random_value1, returns)
```

Comment:
should be done in a for loop that modifies the arrays inplace. You can define a new var that holds all these arrays, and also use it in the loop above

Reply:
You mean something like this? do you have an improvement?

```
            values = [random_value1, random_value2, current_pi_value1, current_pi_value2, next_pi_value1, next_pi_value2]
            for i in range(len(values)):
                values[i] = torch.max(values[i], returns)

            random_value1, random_value2, current_pi_value1, current_pi_value2, next_pi_value1, next_pi_value2 = tuple(values)
```

Reply:
rather 

`torch.max(values, returns, out=values)`

Reply:
If you get issues with arrays and `out`, you can also do 

`values[:] = torch,max(values, returns)`

but my first proposal is more elegant. Python for loops should be avoided as much as possible

Reply:
max() received an invalid combination of arguments - got (list, Tensor), but expected one of:
 * (Tensor input, *, Tensor out)
 * (Tensor input, Tensor other, *, Tensor out)
 * (Tensor input, int dim, bool keepdim, *, tuple of Tensors out)
 * (Tensor input, name dim, bool keepdim, *, tuple of Tensors out)
 
 
This  `values[:] = torch,max(values, returns)` is not working, and I don't think having a for loop really makes the code that elegant since we have to reassign the variables after the for loop. Closing and leaving the code as it is.

Reply:
We misunderstood each other, what I meant was

```
for t in tensors:
   torch.max(t, returns, out=t)
```

It's easy to make mistakes when many names are repeated, and it also makes it a bit harder to read. I can push this myself later if you want to

Reply:
Yes you can go ahead and give it a try, but keep in mind that `torch.max(t, returns, out=t)` doesn't work with tensors that need differentiation.